### PR TITLE
Instructions to use Weave Net for NetworkPolicy

### PR DESCRIPTION
**Pull request by bboreham**
*Description of changes:*

This adds a second set of instructions for NetworkPolicy, using a different Open Source implementation.

Ideally we would clean this up and have a generic set with call-outs to specific parts, but it'll be ugly until we get all the right versions upstream ( also covered by #236 )

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: weavenet/readme.adoc [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+Determine if any of the nodes will require a restart using the command:
+
+  kops rolling-update cluster example.cluster.k8s.local
+
+Output from this command is shown:
+
+```
+$ kops rolling-update cluster example.cluster.k8s.local
+NAME              STATUS      NEEDUPDATE  READY MIN MAX NODES
+master-us-east-1d NeedsUpdate 1           0     1   1   1
```
Comment (arun-gupta) [inline weavenet/readme.adoc]: Existing cluster configuration is 3 master and 5 worker nodes. Can you update the output accordingly?

File: weavenet/readme.adoc [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+Perform the rolling update using the command shown:
+
+  kops rolling-update cluster example.cluster.k8s.local --yes
+
+Output from this command is shown:
+
+```
+NAME              STATUS      NEEDUPDATE  READY MIN MAX NODES
+master-us-east-1d NeedsUpdate 1           0     1   1   1
```
Comment (arun-gupta) [inline weavenet/readme.adoc]: Ditto here

File: weavenet/templates/allow-network-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
```
Comment (arun-gupta) [inline weavenet/templates/allow-network-policy.yaml]: To be consistent, use `apiVersion` as the first line

File: readme.adoc [hunk 4]
```diff
@@ -50,6 +50,7 @@ as you move through the workshop.
 
 | link:statefulsets[StatefulSets with EBS]
 | link:calico[Network policies with Calico]
+| link:weavenet[Network policies with Weave Net]
```
Comment (arun-gupta) [inline readme.adoc]: Create a top-level networking directory, refactor `calico` to that directory. Also move `weavenet` there as well.

**Comment by chrislovecnm** [comment]
Should we have a cni or cni-providers folder rather than weave, calico, the next one? I think it would be cleaner.

File: weavenet/readme.adoc (hunk 1 above)
Comment (bboreham) [reply weavenet/readme.adoc]: This is in the context the user has been [instructed to create a smaller cluster](https://github.com/aws-samples/aws-workshop-for-kubernetes/pull/255/files#diff-0c64f05689728a097473a0fadab32bb2R37) and is now updating it.  That text was copied verbatim from the calico variant.

However #242 points out this update operation is unsupported by `kops`, so I'm re-doing the Weave Net doc to take it out.

**Comment by bboreham** [comment]
@chrislovecnm CNI and NetworkPolicy are separate things, with no real connection.  It happens that the two instances so far in in this tutorial (calico and weave) bundle them in one package, but that's incidental.

I do agree the tutorial could add a CNI section and explain pod network choices, but it won't look like this.

**Comment by bboreham** [comment]
Replaced by #269

**Pull request closed without merging**